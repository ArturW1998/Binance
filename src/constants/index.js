export const ACTION_TYPES = {
  UPDATE_MARKET_PAIRS: 'UPDATE_MARKET_PAIRS',
  SET_ACTIVE_MARKET: 'SET_ACTIVE_MARKET',
  TOGGLE_SOCKET_STREAMS: 'TOGGLE_SOCKET_STREAMS',
};

export const PATHS = {
  BASE_PATH: 'wss://stream.binance.com',
  STREAM_PATH: '/stream',
  STREAM_PARAM: 'streams=',
};

export const LISTS = {
  PAIRS_LIST: ['BNB', 'BTC', 'ETH', 'USDT'],
  DOTS_LIST: [{ id: 1 }, { id: 2 }, { id: 3 }, { id: 4 }],
  TABLE_DATA_LIST: [
    { title: 'Pair' },
    { title: 'Latest Price' },
    { title: 'Open' },
    { title: 'High' },
    { title: 'Low' },
    { title: 'Volume' },
  ],
};

export const CONFIG = {
  BASE_NAME: '/Binance/',
  LOCALSTORAGE_KEY: 'markets',
};
