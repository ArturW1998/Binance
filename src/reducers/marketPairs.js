import { UPDATE_MARKET_PAIRS } from '../constants';

export const initState = {};

const marketPairs = (state = initState, { type, data }) => {
  switch (type) {
    case UPDATE_MARKET_PAIRS:
      return { ...state, ...data };
    default:
      return state;
  }
};

export default marketPairs;
