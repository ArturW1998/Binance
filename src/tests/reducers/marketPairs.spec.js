import { ACTION_TYPES } from '../../constants';
import marketPairs, { initState } from '../../reducers/marketPairs';

describe('marketPairs reducer', () => {
  describe('marketPairs reducer with action', () => {
    const { UPDATE_MARKET_PAIRS } = ACTION_TYPES;

    const action = {
      type: UPDATE_MARKET_PAIRS,
      data: {
        BNBBTC: {},
        ETHUSDT: {},
        ENJBTC: {},
      },
    };

    it(`${UPDATE_MARKET_PAIRS}`, () => {
      const { data } = action;

      const expectedData = {
        ...initState,
        ...data,
      };

      expect(marketPairs(initState, action)).toEqual(expectedData);
    });

    it(`should handle ${UPDATE_MARKET_PAIRS}`, () => {
      expect(marketPairs(initState, action)).toMatchSnapshot();
    });
  });

  describe('marketPairs reducer initial state', () => {
    const action = {};

    it('should return the initial state', () => {
      expect(marketPairs(initState, action)).toEqual(initState);
    });

    it('returns properly', () => {
      expect(marketPairs(initState, action)).toMatchSnapshot();
    });
  });
});
