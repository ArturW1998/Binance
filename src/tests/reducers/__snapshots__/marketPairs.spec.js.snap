// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`marketPairs reducer initial state returns properly 1`] = `Object {}`;

exports[`marketPairs reducer should handle UPDATE_MARKET_PAIRS 1`] = `
Object {
  "BNBBTC": Object {},
  "ENJBTC": Object {},
  "ETHUSDT": Object {},
}
`;
